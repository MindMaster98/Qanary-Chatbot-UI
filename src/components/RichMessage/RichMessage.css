.richContentContainer {
    width: 100%;
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
}

.richContentContainer .button {
    min-width: 100px;
    min-height: 2.1rem;
    padding: 0 1.1em;
    margin: 16px;
}

.richContentContainer .button:hover {
    transition: background 0.2s;
    background-color: #bd818d;
}

.buttonsContainer, .diagramContainer {
    width: 100%;
    display: flex;
    justify-content: center;
    margin-bottom: 2vh;
}