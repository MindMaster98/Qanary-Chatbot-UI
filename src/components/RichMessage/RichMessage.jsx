import React from "react";
import { Button, Message } from "..";
import "./RichMessage.css";

/**
 * @typedef {object} RichMessage
 *   @property {boolean} isReply determines whether it's a bot or user message (left or right side, colour scheme, etc.)
 *   @property {boolean} loadedSuccessfully determines if source disclaimer gets displayed or not
 *   @property {boolean} followUpNeeded determines if extra buttons are displayed
 *   @property {string} time The time the message was sent
 *   @property {string} icon The image next to the message
 *   @property {string} text The message's contents
 *   @property {object} visualization data on how the visualizations, i. e. buttons, look
 *
 * @description A rich content variant of the message component, supporting feedback through buttons
 *
 * @property {RichMessage} messageObject
 */
export default function RichMessage(props) {
  const buttonElements = !!props.messageObject?.visualization?.buttons?.map
    ? props.messageObject.visualization.buttons.map((button, i) => (
        // i is the shortest unique identifier in this case and the content will not be updated
        <Button
          text={button.title ?? ""}
          key={i}
          style={{
            minWidth: `${100}px`,
            height: `${35}px`,
          }}
          onClick={button.onClick ?? undefined}
        />
      ))
    : null;

  return (
    <div tabIndex="-1">
      <Message messageObject={props.messageObject} />
      <div tabIndex="-1" className="richContentContainer">
        <div tabIndex="-1" className="buttonsContainer">
          {buttonElements}
        </div>
      </div>
    </div>
  );
}
