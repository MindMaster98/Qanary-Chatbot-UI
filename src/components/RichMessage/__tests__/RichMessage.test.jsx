import renderer from "react-test-renderer";
import RichMessage from "../RichMessage";

// Snapshot tests
// Message and Button specific tests are obsolete since they're already covered by the Message and Button components' tests

it("renders correctly with no props", () => {
  const tree = renderer.create(<RichMessage />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders one button with single valid button visualization as input", () => {
  const tree = renderer
    .create(
      <RichMessage
        messageObject={{
          visualization: {
            buttons: [
              {
                title: "test",
              },
            ],
          },
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders two distinct buttons with two valid, distinct button visualization as input", () => {
  const tree = renderer
    .create(
      <RichMessage
        messageObject={{
          visualization: {
            buttons: [
              {
                title: "test",
              },
              {
                title: "test2",
              },
            ],
          },
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders two distinct buttons with two valid, equal button visualizations as input", () => {
  const tree = renderer
    .create(
      <RichMessage
        messageObject={{
          visualization: {
            buttons: [
              {
                title: "test",
              },
              {
                title: "test",
              },
            ],
          },
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders no button with single invalid button visualization", () => {
  const tree = renderer
    .create(
      <RichMessage
        messageObject={{
          visualization: {
            buttons: {
              title: "test",
            },
          },
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
