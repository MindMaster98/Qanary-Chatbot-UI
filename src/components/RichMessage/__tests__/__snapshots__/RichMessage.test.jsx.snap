// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with no props 1`] = `
<div
  tabIndex="-1"
>
  <div
    className="messageContainer"
    tabIndex="-1"
  >
    <div
      className="clickableIcon "
      style={
        Object {
          "cursor": "initial",
          "height": "100px",
          "width": "100px",
        }
      }
      tabIndex="-1"
    >
      <img
        alt="Nutzer"
        src="account_icon_black.svg"
        tabIndex="-1"
      />
    </div>
    <div
      className="messageDataContainer"
      tabIndex="0"
    >
      <div
        className="messageText black-text"
        dangerouslySetInnerHTML={
          Object {
            "__html": "",
          }
        }
      />
      <div
        className="messageTimeSent"
      />
    </div>
  </div>
  <div
    className="richContentContainer"
    tabIndex="-1"
  >
    <div
      className="buttonsContainer"
      tabIndex="-1"
    />
  </div>
</div>
`;

exports[`renders no button with single invalid button visualization 1`] = `
<div
  tabIndex="-1"
>
  <div
    className="messageContainer"
    tabIndex="-1"
  >
    <div
      className="clickableIcon "
      style={
        Object {
          "cursor": "initial",
          "height": "100px",
          "width": "100px",
        }
      }
      tabIndex="-1"
    >
      <img
        alt="Nutzer"
        src="account_icon_black.svg"
        tabIndex="-1"
      />
    </div>
    <div
      className="messageDataContainer"
      tabIndex="0"
    >
      <div
        className="messageText black-text"
        dangerouslySetInnerHTML={
          Object {
            "__html": "",
          }
        }
      />
      <div
        className="messageTimeSent"
      />
    </div>
  </div>
  <div
    className="richContentContainer"
    tabIndex="-1"
  >
    <div
      className="buttonsContainer"
      tabIndex="-1"
    />
  </div>
</div>
`;

exports[`renders one button with single valid button visualization as input 1`] = `
<div
  tabIndex="-1"
>
  <div
    className="messageContainer"
    tabIndex="-1"
  >
    <div
      className="clickableIcon "
      style={
        Object {
          "cursor": "initial",
          "height": "100px",
          "width": "100px",
        }
      }
      tabIndex="-1"
    >
      <img
        alt="Nutzer"
        src="account_icon_black.svg"
        tabIndex="-1"
      />
    </div>
    <div
      className="messageDataContainer"
      tabIndex="0"
    >
      <div
        className="messageText black-text"
        dangerouslySetInnerHTML={
          Object {
            "__html": "",
          }
        }
      />
      <div
        className="messageTimeSent"
      />
    </div>
  </div>
  <div
    className="richContentContainer"
    tabIndex="-1"
  >
    <div
      className="buttonsContainer"
      tabIndex="-1"
    >
      <button
        className="button "
        style={
          Object {
            "height": "35px",
            "minWidth": "100px",
          }
        }
      >
        test
      </button>
    </div>
  </div>
</div>
`;

exports[`renders two distinct buttons with two valid, distinct button visualization as input 1`] = `
<div
  tabIndex="-1"
>
  <div
    className="messageContainer"
    tabIndex="-1"
  >
    <div
      className="clickableIcon "
      style={
        Object {
          "cursor": "initial",
          "height": "100px",
          "width": "100px",
        }
      }
      tabIndex="-1"
    >
      <img
        alt="Nutzer"
        src="account_icon_black.svg"
        tabIndex="-1"
      />
    </div>
    <div
      className="messageDataContainer"
      tabIndex="0"
    >
      <div
        className="messageText black-text"
        dangerouslySetInnerHTML={
          Object {
            "__html": "",
          }
        }
      />
      <div
        className="messageTimeSent"
      />
    </div>
  </div>
  <div
    className="richContentContainer"
    tabIndex="-1"
  >
    <div
      className="buttonsContainer"
      tabIndex="-1"
    >
      <button
        className="button "
        style={
          Object {
            "height": "35px",
            "minWidth": "100px",
          }
        }
      >
        test
      </button>
      <button
        className="button "
        style={
          Object {
            "height": "35px",
            "minWidth": "100px",
          }
        }
      >
        test2
      </button>
    </div>
  </div>
</div>
`;

exports[`renders two distinct buttons with two valid, equal button visualizations as input 1`] = `
<div
  tabIndex="-1"
>
  <div
    className="messageContainer"
    tabIndex="-1"
  >
    <div
      className="clickableIcon "
      style={
        Object {
          "cursor": "initial",
          "height": "100px",
          "width": "100px",
        }
      }
      tabIndex="-1"
    >
      <img
        alt="Nutzer"
        src="account_icon_black.svg"
        tabIndex="-1"
      />
    </div>
    <div
      className="messageDataContainer"
      tabIndex="0"
    >
      <div
        className="messageText black-text"
        dangerouslySetInnerHTML={
          Object {
            "__html": "",
          }
        }
      />
      <div
        className="messageTimeSent"
      />
    </div>
  </div>
  <div
    className="richContentContainer"
    tabIndex="-1"
  >
    <div
      className="buttonsContainer"
      tabIndex="-1"
    >
      <button
        className="button "
        style={
          Object {
            "height": "35px",
            "minWidth": "100px",
          }
        }
      >
        test
      </button>
      <button
        className="button "
        style={
          Object {
            "height": "35px",
            "minWidth": "100px",
          }
        }
      >
        test
      </button>
    </div>
  </div>
</div>
`;
